### DISCUSSION TITLE
I cant't get the latest cookies via middleware or nuxtServerInit in server side.

### DISCUSSION THREAD

**Discussion by kratospan** [nuxt/nuxt#21949] [Questions (Nuxt 2)]
When a user logs in, I set a token in cookies. I have written a middleware to check whether the user is logged in or not based on the token. However, I have noticed that when the page is first loaded, the cookies I obtain from context.req are not the latest version. Only when I refresh the page, do I get the updated cookies.

So, if I want to obtain the latest cookies on the server-side, what should I do?

**Comment by pont1s** [comment]
We have the same problem. We use useCookie to set the value of the JWT token. When the page is rendered on the server side and the token has expired, after updating the value via useCookie, the old value remains in new requests.
